src/diffusers/models/model_loading_utils.py

```diff
@@ -99,6 +131,8 @@ def load_state_dict(checkpoint_file: Union[str, os.PathLike], variant: Optional[
     """
     Reads a checkpoint file, returning properly formatted errors if they arise.
     """
+    if isinstance(checkpoint_file, dict):
```

Comment:
We merge the sharded checkpoints (as stated in the PR description and mutually agreed upon internally) in case we're doing quantization:

https://github.com/huggingface/diffusers/blob/55f96d8ee5793d56f4073ffbf5c47baec3d1e401/src/diffusers/models/modeling_utils.py#L765

^ `model_file` becomes a state dict which is loaded by `load_state_dict`:

https://github.com/huggingface/diffusers/blob/55f96d8ee5793d56f4073ffbf5c47baec3d1e401/src/diffusers/models/modeling_utils.py#L836

and hence this change.

Reply:
71316a663